xdsl/dialects/riscv.py

```diff
@@ -0,0 +1,756 @@
+from __future__ import annotations
+from dataclasses import dataclass, field
+
+from typing import Union, TypeVar, Type
+
+from xdsl.ir import MLContext, Operation, Data
+from xdsl.irdl import irdl_op_definition, irdl_attr_definition, builder, AnyOf
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+from xdsl.dialects.builtin import StringAttr, AttributeDef, IntegerAttr
+
+
+@dataclass(frozen=True)
+class Register:
+    """A riscv register."""
+
+    index: int = field()
+    """The register index. Can be between 0 and 31."""
+
+    abi_names = {
+        "zero": 0,
+        "ra": 1,
+        "sp": 2,
+        "gp": 3,
+        "tp": 4,
+        "t0": 5,
+        "t1": 6,
+        "t2": 7,
+        "fp": 8,
+        "s0": 8,
+        "s1": 9,
+        "a0": 10,
+        "a1": 11,
+        "a2": 12,
+        "a3": 13,
+        "a4": 14,
+        "a5": 15,
+        "a6": 16,
+        "a7": 17,
+        "s2": 18,
+        "s3": 19,
+        "s4": 20,
+        "s5": 21,
+        "s6": 22,
+        "s7": 23,
+        "s8": 24,
+        "s9": 25,
+        "s10": 26,
+        "s11": 27,
+        "t3": 28,
+        "t4": 29,
+        "t5": 30,
+        "t6": 31
+    }
+
+    @staticmethod
+    def from_index(index: int) -> Register:
+        assert 32 > index >= 0
+        register = Register(index)
+        return register
+
+    @staticmethod
+    def from_name(name: str) -> Register:
+        if name in Register.abi_names:
+            return Register.from_index(Register.abi_names[name])
+        if name[0] == 'x' and name[1:].isnumeric():
+            return Register.from_index(int(name[1:]))
+        assert False and "register with unknown name"
```

Comment:
that's a funky syntax, why not the following?
```suggestion
        assert False, "register with unknown name"
```

Reply:
I have no clue, this is copy-pasta, I'll apply your syntax. Thanks!